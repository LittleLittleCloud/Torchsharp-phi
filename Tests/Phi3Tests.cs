using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static TorchSharp.torch;
using TorchSharp;
using Xunit;
using FluentAssertions;

namespace Phi.Tests;

public class Phi3Tests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public async Task ShapeTest()
    {
        var modelWeightFolder = "C:\\Users\\xiaoyuz\\source\\repos\\Phi-3-mini-4k-instruct";
        var model = Phi3ForCasualLM.FromPretrained(modelWeightFolder, torchDtype: ScalarType.BFloat16);
        var state_dict_str = model.Peek();
        Approvals.Verify(state_dict_str);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public async Task TokenizerTest()
    {
        var modelWeightFolder = "C:\\Users\\xiaoyuz\\source\\repos\\Phi-3-mini-4k-instruct";
        var tokenizer = LLama2Tokenizer.FromPretrained(modelWeightFolder);
        tokenizer.BosId.Should().Be(1);
        tokenizer.EosId.Should().Be(32000);
        var messages = new string[]
        {
            "Can you provide ways to eat combinations of bananas and dragonfruits?",
            "Sure! Here are some ways to eat bananas and dragonfruits together: 1. Banana and dragonfruit smoothie: Blend bananas and dragonfruits together with some milk and honey. 2. Banana and dragonfruit salad: Mix sliced bananas and dragonfruits together with some lemon juice and honey.",
            "What about solving an 2x + 3 = 7 equation?"
        };
        var sb = new StringBuilder();
        foreach (var message in messages)
        {
            var tokenized = tokenizer.Encode(message, true, false);
            var tokenized_str = string.Join(", ", tokenized.Select(x => x.ToString()));

            sb.AppendLine(tokenized_str);
        }
        Approvals.Verify(sb.ToString());
    }
}
