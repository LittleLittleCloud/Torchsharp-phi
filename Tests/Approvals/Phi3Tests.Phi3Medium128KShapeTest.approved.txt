0: lm_head.weight: sum: 8831.63  dtype: BFloat16 shape: [32064,5120]
1: model.embed_tokens.weight: sum: 9089.119  dtype: BFloat16 shape: [32064,5120]
2: model.layers.0.input_layernorm.weight: sum: 47.5925  dtype: BFloat16 shape: [5120]
3: model.layers.0.mlp.down_proj.weight: sum: 7163.753  dtype: BFloat16 shape: [5120,17920]
4: model.layers.0.mlp.gate_up_proj.weight: sum: 6331.9746  dtype: BFloat16 shape: [35840,5120]
5: model.layers.0.post_attention_layernorm.weight: sum: 47.9208  dtype: BFloat16 shape: [5120]
6: model.layers.0.self_attn.o_proj.weight: sum: 3848.5532  dtype: BFloat16 shape: [5120,5120]
7: model.layers.0.self_attn.qkv_proj.weight: sum: 8421.841  dtype: BFloat16 shape: [7680,5120]
8: model.layers.1.input_layernorm.weight: sum: 48.6108  dtype: BFloat16 shape: [5120]
9: model.layers.1.mlp.down_proj.weight: sum: 8425.917  dtype: BFloat16 shape: [5120,17920]
10: model.layers.1.mlp.gate_up_proj.weight: sum: 3712.3936  dtype: BFloat16 shape: [35840,5120]
11: model.layers.1.post_attention_layernorm.weight: sum: 47.5735  dtype: BFloat16 shape: [5120]
12: model.layers.1.self_attn.o_proj.weight: sum: 3584.868  dtype: BFloat16 shape: [5120,5120]
13: model.layers.1.self_attn.qkv_proj.weight: sum: -1851.4304  dtype: BFloat16 shape: [7680,5120]
14: model.layers.10.input_layernorm.weight: sum: 47.6563  dtype: BFloat16 shape: [5120]
15: model.layers.10.mlp.down_proj.weight: sum: 18606.283  dtype: BFloat16 shape: [5120,17920]
16: model.layers.10.mlp.gate_up_proj.weight: sum: 5648.0195  dtype: BFloat16 shape: [35840,5120]
17: model.layers.10.post_attention_layernorm.weight: sum: 47.6913  dtype: BFloat16 shape: [5120]
18: model.layers.10.self_attn.o_proj.weight: sum: -10016.966  dtype: BFloat16 shape: [5120,5120]
19: model.layers.10.self_attn.qkv_proj.weight: sum: 16551.19  dtype: BFloat16 shape: [7680,5120]
20: model.layers.11.input_layernorm.weight: sum: 47.6281  dtype: BFloat16 shape: [5120]
21: model.layers.11.mlp.down_proj.weight: sum: 13680.15  dtype: BFloat16 shape: [5120,17920]
22: model.layers.11.mlp.gate_up_proj.weight: sum: 7461.1055  dtype: BFloat16 shape: [35840,5120]
23: model.layers.11.post_attention_layernorm.weight: sum: 47.6767  dtype: BFloat16 shape: [5120]
24: model.layers.11.self_attn.o_proj.weight: sum: 3205.6362  dtype: BFloat16 shape: [5120,5120]
25: model.layers.11.self_attn.qkv_proj.weight: sum: 6098.9897  dtype: BFloat16 shape: [7680,5120]
26: model.layers.12.input_layernorm.weight: sum: 47.5975  dtype: BFloat16 shape: [5120]
27: model.layers.12.mlp.down_proj.weight: sum: 4160.7534  dtype: BFloat16 shape: [5120,17920]
28: model.layers.12.mlp.gate_up_proj.weight: sum: 10259.703  dtype: BFloat16 shape: [35840,5120]
29: model.layers.12.post_attention_layernorm.weight: sum: 47.6839  dtype: BFloat16 shape: [5120]
30: model.layers.12.self_attn.o_proj.weight: sum: -144.5138  dtype: BFloat16 shape: [5120,5120]
31: model.layers.12.self_attn.qkv_proj.weight: sum: 7577.6426  dtype: BFloat16 shape: [7680,5120]
32: model.layers.13.input_layernorm.weight: sum: 47.6283  dtype: BFloat16 shape: [5120]
33: model.layers.13.mlp.down_proj.weight: sum: 4671.2065  dtype: BFloat16 shape: [5120,17920]
34: model.layers.13.mlp.gate_up_proj.weight: sum: 9677.234  dtype: BFloat16 shape: [35840,5120]
35: model.layers.13.post_attention_layernorm.weight: sum: 47.6188  dtype: BFloat16 shape: [5120]
36: model.layers.13.self_attn.o_proj.weight: sum: 2640.277  dtype: BFloat16 shape: [5120,5120]
37: model.layers.13.self_attn.qkv_proj.weight: sum: 3878.5913  dtype: BFloat16 shape: [7680,5120]
38: model.layers.14.input_layernorm.weight: sum: 47.7102  dtype: BFloat16 shape: [5120]
39: model.layers.14.mlp.down_proj.weight: sum: 2581.2864  dtype: BFloat16 shape: [5120,17920]
40: model.layers.14.mlp.gate_up_proj.weight: sum: 5908.0703  dtype: BFloat16 shape: [35840,5120]
41: model.layers.14.post_attention_layernorm.weight: sum: 47.713  dtype: BFloat16 shape: [5120]
42: model.layers.14.self_attn.o_proj.weight: sum: 2888.0823  dtype: BFloat16 shape: [5120,5120]
43: model.layers.14.self_attn.qkv_proj.weight: sum: 5127.94  dtype: BFloat16 shape: [7680,5120]
44: model.layers.15.input_layernorm.weight: sum: 47.8132  dtype: BFloat16 shape: [5120]
45: model.layers.15.mlp.down_proj.weight: sum: 8607.194  dtype: BFloat16 shape: [5120,17920]
46: model.layers.15.mlp.gate_up_proj.weight: sum: -106967.695  dtype: BFloat16 shape: [35840,5120]
47: model.layers.15.post_attention_layernorm.weight: sum: 47.706  dtype: BFloat16 shape: [5120]
48: model.layers.15.self_attn.o_proj.weight: sum: 3744.1333  dtype: BFloat16 shape: [5120,5120]
49: model.layers.15.self_attn.qkv_proj.weight: sum: 5998.7236  dtype: BFloat16 shape: [7680,5120]
50: model.layers.16.input_layernorm.weight: sum: 47.6402  dtype: BFloat16 shape: [5120]
51: model.layers.16.mlp.down_proj.weight: sum: 21114.156  dtype: BFloat16 shape: [5120,17920]
52: model.layers.16.mlp.gate_up_proj.weight: sum: -7858.0854  dtype: BFloat16 shape: [35840,5120]
53: model.layers.16.post_attention_layernorm.weight: sum: 47.7447  dtype: BFloat16 shape: [5120]
54: model.layers.16.self_attn.o_proj.weight: sum: 2907.1401  dtype: BFloat16 shape: [5120,5120]
55: model.layers.16.self_attn.qkv_proj.weight: sum: 6099.6587  dtype: BFloat16 shape: [7680,5120]
56: model.layers.17.input_layernorm.weight: sum: 47.6845  dtype: BFloat16 shape: [5120]
57: model.layers.17.mlp.down_proj.weight: sum: 18616.076  dtype: BFloat16 shape: [5120,17920]
58: model.layers.17.mlp.gate_up_proj.weight: sum: -9960.982  dtype: BFloat16 shape: [35840,5120]
59: model.layers.17.post_attention_layernorm.weight: sum: 47.5646  dtype: BFloat16 shape: [5120]
60: model.layers.17.self_attn.o_proj.weight: sum: 2699.6113  dtype: BFloat16 shape: [5120,5120]
61: model.layers.17.self_attn.qkv_proj.weight: sum: 5455.515  dtype: BFloat16 shape: [7680,5120]
62: model.layers.18.input_layernorm.weight: sum: 47.65  dtype: BFloat16 shape: [5120]
63: model.layers.18.mlp.down_proj.weight: sum: 5474.8877  dtype: BFloat16 shape: [5120,17920]
64: model.layers.18.mlp.gate_up_proj.weight: sum: 4656.6816  dtype: BFloat16 shape: [35840,5120]
65: model.layers.18.post_attention_layernorm.weight: sum: 47.7978  dtype: BFloat16 shape: [5120]
66: model.layers.18.self_attn.o_proj.weight: sum: 4761.012  dtype: BFloat16 shape: [5120,5120]
67: model.layers.18.self_attn.qkv_proj.weight: sum: 625.1904  dtype: BFloat16 shape: [7680,5120]
68: model.layers.19.input_layernorm.weight: sum: 47.7823  dtype: BFloat16 shape: [5120]
69: model.layers.19.mlp.down_proj.weight: sum: 5780.411  dtype: BFloat16 shape: [5120,17920]
70: model.layers.19.mlp.gate_up_proj.weight: sum: 5403.0503  dtype: BFloat16 shape: [35840,5120]
71: model.layers.19.post_attention_layernorm.weight: sum: 47.7153  dtype: BFloat16 shape: [5120]
72: model.layers.19.self_attn.o_proj.weight: sum: 4119.3535  dtype: BFloat16 shape: [5120,5120]
73: model.layers.19.self_attn.qkv_proj.weight: sum: 4126.5503  dtype: BFloat16 shape: [7680,5120]
74: model.layers.2.input_layernorm.weight: sum: 47.8054  dtype: BFloat16 shape: [5120]
75: model.layers.2.mlp.down_proj.weight: sum: -10342.353  dtype: BFloat16 shape: [5120,17920]
76: model.layers.2.mlp.gate_up_proj.weight: sum: 6488.002  dtype: BFloat16 shape: [35840,5120]
77: model.layers.2.post_attention_layernorm.weight: sum: 47.5547  dtype: BFloat16 shape: [5120]
78: model.layers.2.self_attn.o_proj.weight: sum: 5156.435  dtype: BFloat16 shape: [5120,5120]
79: model.layers.2.self_attn.qkv_proj.weight: sum: 4946.5  dtype: BFloat16 shape: [7680,5120]
80: model.layers.20.input_layernorm.weight: sum: 47.6376  dtype: BFloat16 shape: [5120]
81: model.layers.20.mlp.down_proj.weight: sum: 7200.665  dtype: BFloat16 shape: [5120,17920]
82: model.layers.20.mlp.gate_up_proj.weight: sum: 7013.092  dtype: BFloat16 shape: [35840,5120]
83: model.layers.20.post_attention_layernorm.weight: sum: 47.5653  dtype: BFloat16 shape: [5120]
84: model.layers.20.self_attn.o_proj.weight: sum: 6396.3843  dtype: BFloat16 shape: [5120,5120]
85: model.layers.20.self_attn.qkv_proj.weight: sum: -2300.7656  dtype: BFloat16 shape: [7680,5120]
86: model.layers.21.input_layernorm.weight: sum: 47.6506  dtype: BFloat16 shape: [5120]
87: model.layers.21.mlp.down_proj.weight: sum: 9838.712  dtype: BFloat16 shape: [5120,17920]
88: model.layers.21.mlp.gate_up_proj.weight: sum: 8324.952  dtype: BFloat16 shape: [35840,5120]
89: model.layers.21.post_attention_layernorm.weight: sum: 47.6261  dtype: BFloat16 shape: [5120]
90: model.layers.21.self_attn.o_proj.weight: sum: 1822.6504  dtype: BFloat16 shape: [5120,5120]
91: model.layers.21.self_attn.qkv_proj.weight: sum: 1820.4865  dtype: BFloat16 shape: [7680,5120]
92: model.layers.22.input_layernorm.weight: sum: 47.5751  dtype: BFloat16 shape: [5120]
93: model.layers.22.mlp.down_proj.weight: sum: 5612.3  dtype: BFloat16 shape: [5120,17920]
94: model.layers.22.mlp.gate_up_proj.weight: sum: 6000.775  dtype: BFloat16 shape: [35840,5120]
95: model.layers.22.post_attention_layernorm.weight: sum: 47.635  dtype: BFloat16 shape: [5120]
96: model.layers.22.self_attn.o_proj.weight: sum: 3235.1814  dtype: BFloat16 shape: [5120,5120]
97: model.layers.22.self_attn.qkv_proj.weight: sum: 6110.013  dtype: BFloat16 shape: [7680,5120]
98: model.layers.23.input_layernorm.weight: sum: 47.7673  dtype: BFloat16 shape: [5120]
99: model.layers.23.mlp.down_proj.weight: sum: 6398.807  dtype: BFloat16 shape: [5120,17920]
100: model.layers.23.mlp.gate_up_proj.weight: sum: 16104.477  dtype: BFloat16 shape: [35840,5120]
101: model.layers.23.post_attention_layernorm.weight: sum: 47.7493  dtype: BFloat16 shape: [5120]
102: model.layers.23.self_attn.o_proj.weight: sum: 4702.067  dtype: BFloat16 shape: [5120,5120]
103: model.layers.23.self_attn.qkv_proj.weight: sum: 3266.9082  dtype: BFloat16 shape: [7680,5120]
104: model.layers.24.input_layernorm.weight: sum: 47.6928  dtype: BFloat16 shape: [5120]
105: model.layers.24.mlp.down_proj.weight: sum: 8130.848  dtype: BFloat16 shape: [5120,17920]
106: model.layers.24.mlp.gate_up_proj.weight: sum: 6531.7915  dtype: BFloat16 shape: [35840,5120]
107: model.layers.24.post_attention_layernorm.weight: sum: 47.595  dtype: BFloat16 shape: [5120]
108: model.layers.24.self_attn.o_proj.weight: sum: 2620.3318  dtype: BFloat16 shape: [5120,5120]
109: model.layers.24.self_attn.qkv_proj.weight: sum: 5599.595  dtype: BFloat16 shape: [7680,5120]
110: model.layers.25.input_layernorm.weight: sum: 47.6321  dtype: BFloat16 shape: [5120]
111: model.layers.25.mlp.down_proj.weight: sum: 9211.291  dtype: BFloat16 shape: [5120,17920]
112: model.layers.25.mlp.gate_up_proj.weight: sum: 7608.4526  dtype: BFloat16 shape: [35840,5120]
113: model.layers.25.post_attention_layernorm.weight: sum: 47.7085  dtype: BFloat16 shape: [5120]
114: model.layers.25.self_attn.o_proj.weight: sum: 7783.242  dtype: BFloat16 shape: [5120,5120]
115: model.layers.25.self_attn.qkv_proj.weight: sum: 7869.7383  dtype: BFloat16 shape: [7680,5120]
116: model.layers.26.input_layernorm.weight: sum: 47.6653  dtype: BFloat16 shape: [5120]
117: model.layers.26.mlp.down_proj.weight: sum: 7981.1343  dtype: BFloat16 shape: [5120,17920]
118: model.layers.26.mlp.gate_up_proj.weight: sum: 6676.15  dtype: BFloat16 shape: [35840,5120]
119: model.layers.26.post_attention_layernorm.weight: sum: 47.525  dtype: BFloat16 shape: [5120]
120: model.layers.26.self_attn.o_proj.weight: sum: 4953.31  dtype: BFloat16 shape: [5120,5120]
121: model.layers.26.self_attn.qkv_proj.weight: sum: 3810.552  dtype: BFloat16 shape: [7680,5120]
122: model.layers.27.input_layernorm.weight: sum: 47.7611  dtype: BFloat16 shape: [5120]
123: model.layers.27.mlp.down_proj.weight: sum: 23738.281  dtype: BFloat16 shape: [5120,17920]
124: model.layers.27.mlp.gate_up_proj.weight: sum: 15378.64  dtype: BFloat16 shape: [35840,5120]
125: model.layers.27.post_attention_layernorm.weight: sum: 47.693  dtype: BFloat16 shape: [5120]
126: model.layers.27.self_attn.o_proj.weight: sum: 2521.0718  dtype: BFloat16 shape: [5120,5120]
127: model.layers.27.self_attn.qkv_proj.weight: sum: 3758.5137  dtype: BFloat16 shape: [7680,5120]
128: model.layers.28.input_layernorm.weight: sum: 47.6298  dtype: BFloat16 shape: [5120]
129: model.layers.28.mlp.down_proj.weight: sum: 7606.8823  dtype: BFloat16 shape: [5120,17920]
130: model.layers.28.mlp.gate_up_proj.weight: sum: 7802.421  dtype: BFloat16 shape: [35840,5120]
131: model.layers.28.post_attention_layernorm.weight: sum: 47.7415  dtype: BFloat16 shape: [5120]
132: model.layers.28.self_attn.o_proj.weight: sum: 2160.5444  dtype: BFloat16 shape: [5120,5120]
133: model.layers.28.self_attn.qkv_proj.weight: sum: 3768.0613  dtype: BFloat16 shape: [7680,5120]
134: model.layers.29.input_layernorm.weight: sum: 47.6656  dtype: BFloat16 shape: [5120]
135: model.layers.29.mlp.down_proj.weight: sum: 43169.176  dtype: BFloat16 shape: [5120,17920]
136: model.layers.29.mlp.gate_up_proj.weight: sum: 6853.7383  dtype: BFloat16 shape: [35840,5120]
137: model.layers.29.post_attention_layernorm.weight: sum: 47.5453  dtype: BFloat16 shape: [5120]
138: model.layers.29.self_attn.o_proj.weight: sum: 3383.2505  dtype: BFloat16 shape: [5120,5120]
139: model.layers.29.self_attn.qkv_proj.weight: sum: 9114.548  dtype: BFloat16 shape: [7680,5120]
140: model.layers.3.input_layernorm.weight: sum: 47.7772  dtype: BFloat16 shape: [5120]
141: model.layers.3.mlp.down_proj.weight: sum: 6667.656  dtype: BFloat16 shape: [5120,17920]
142: model.layers.3.mlp.gate_up_proj.weight: sum: 7334.932  dtype: BFloat16 shape: [35840,5120]
143: model.layers.3.post_attention_layernorm.weight: sum: 47.6348  dtype: BFloat16 shape: [5120]
144: model.layers.3.self_attn.o_proj.weight: sum: 3615.2449  dtype: BFloat16 shape: [5120,5120]
145: model.layers.3.self_attn.qkv_proj.weight: sum: 2604.8662  dtype: BFloat16 shape: [7680,5120]
146: model.layers.30.input_layernorm.weight: sum: 47.8029  dtype: BFloat16 shape: [5120]
147: model.layers.30.mlp.down_proj.weight: sum: 11730.741  dtype: BFloat16 shape: [5120,17920]
148: model.layers.30.mlp.gate_up_proj.weight: sum: 5316.168  dtype: BFloat16 shape: [35840,5120]
149: model.layers.30.post_attention_layernorm.weight: sum: 47.6442  dtype: BFloat16 shape: [5120]
150: model.layers.30.self_attn.o_proj.weight: sum: 2667.7178  dtype: BFloat16 shape: [5120,5120]
151: model.layers.30.self_attn.qkv_proj.weight: sum: -66172.805  dtype: BFloat16 shape: [7680,5120]
152: model.layers.31.input_layernorm.weight: sum: 47.7874  dtype: BFloat16 shape: [5120]
153: model.layers.31.mlp.down_proj.weight: sum: 16029.862  dtype: BFloat16 shape: [5120,17920]
154: model.layers.31.mlp.gate_up_proj.weight: sum: 12025.813  dtype: BFloat16 shape: [35840,5120]
155: model.layers.31.post_attention_layernorm.weight: sum: 47.7632  dtype: BFloat16 shape: [5120]
156: model.layers.31.self_attn.o_proj.weight: sum: 4165.232  dtype: BFloat16 shape: [5120,5120]
157: model.layers.31.self_attn.qkv_proj.weight: sum: 6578.292  dtype: BFloat16 shape: [7680,5120]
158: model.layers.32.input_layernorm.weight: sum: 47.5938  dtype: BFloat16 shape: [5120]
159: model.layers.32.mlp.down_proj.weight: sum: 3180.2656  dtype: BFloat16 shape: [5120,17920]
160: model.layers.32.mlp.gate_up_proj.weight: sum: 13004.181  dtype: BFloat16 shape: [35840,5120]
161: model.layers.32.post_attention_layernorm.weight: sum: 47.6422  dtype: BFloat16 shape: [5120]
162: model.layers.32.self_attn.o_proj.weight: sum: 2407.591  dtype: BFloat16 shape: [5120,5120]
163: model.layers.32.self_attn.qkv_proj.weight: sum: 5231.5703  dtype: BFloat16 shape: [7680,5120]
164: model.layers.33.input_layernorm.weight: sum: 47.6873  dtype: BFloat16 shape: [5120]
165: model.layers.33.mlp.down_proj.weight: sum: -102888.96  dtype: BFloat16 shape: [5120,17920]
166: model.layers.33.mlp.gate_up_proj.weight: sum: 10202.163  dtype: BFloat16 shape: [35840,5120]
167: model.layers.33.post_attention_layernorm.weight: sum: 47.789  dtype: BFloat16 shape: [5120]
168: model.layers.33.self_attn.o_proj.weight: sum: 2555.2312  dtype: BFloat16 shape: [5120,5120]
169: model.layers.33.self_attn.qkv_proj.weight: sum: 4159.513  dtype: BFloat16 shape: [7680,5120]
170: model.layers.34.input_layernorm.weight: sum: 47.7731  dtype: BFloat16 shape: [5120]
171: model.layers.34.mlp.down_proj.weight: sum: 7630.083  dtype: BFloat16 shape: [5120,17920]
172: model.layers.34.mlp.gate_up_proj.weight: sum: 6803.9946  dtype: BFloat16 shape: [35840,5120]
173: model.layers.34.post_attention_layernorm.weight: sum: 47.5841  dtype: BFloat16 shape: [5120]
174: model.layers.34.self_attn.o_proj.weight: sum: 3108.2104  dtype: BFloat16 shape: [5120,5120]
175: model.layers.34.self_attn.qkv_proj.weight: sum: -190.0064  dtype: BFloat16 shape: [7680,5120]
176: model.layers.35.input_layernorm.weight: sum: 47.687  dtype: BFloat16 shape: [5120]
177: model.layers.35.mlp.down_proj.weight: sum: 6784.7754  dtype: BFloat16 shape: [5120,17920]
178: model.layers.35.mlp.gate_up_proj.weight: sum: 4955.9663  dtype: BFloat16 shape: [35840,5120]
179: model.layers.35.post_attention_layernorm.weight: sum: 47.7154  dtype: BFloat16 shape: [5120]
180: model.layers.35.self_attn.o_proj.weight: sum: 585.6212  dtype: BFloat16 shape: [5120,5120]
181: model.layers.35.self_attn.qkv_proj.weight: sum: 11762.182  dtype: BFloat16 shape: [7680,5120]
182: model.layers.36.input_layernorm.weight: sum: 47.6995  dtype: BFloat16 shape: [5120]
183: model.layers.36.mlp.down_proj.weight: sum: 4884.1885  dtype: BFloat16 shape: [5120,17920]
184: model.layers.36.mlp.gate_up_proj.weight: sum: -76.9171  dtype: BFloat16 shape: [35840,5120]
185: model.layers.36.post_attention_layernorm.weight: sum: 47.6453  dtype: BFloat16 shape: [5120]
186: model.layers.36.self_attn.o_proj.weight: sum: 1539.1879  dtype: BFloat16 shape: [5120,5120]
187: model.layers.36.self_attn.qkv_proj.weight: sum: 5276.8296  dtype: BFloat16 shape: [7680,5120]
188: model.layers.37.input_layernorm.weight: sum: 47.6963  dtype: BFloat16 shape: [5120]
189: model.layers.37.mlp.down_proj.weight: sum: 8895.856  dtype: BFloat16 shape: [5120,17920]
190: model.layers.37.mlp.gate_up_proj.weight: sum: 6540.262  dtype: BFloat16 shape: [35840,5120]
191: model.layers.37.post_attention_layernorm.weight: sum: 47.6787  dtype: BFloat16 shape: [5120]
192: model.layers.37.self_attn.o_proj.weight: sum: 670.6048  dtype: BFloat16 shape: [5120,5120]
193: model.layers.37.self_attn.qkv_proj.weight: sum: -1171.039  dtype: BFloat16 shape: [7680,5120]
194: model.layers.38.input_layernorm.weight: sum: 47.6971  dtype: BFloat16 shape: [5120]
195: model.layers.38.mlp.down_proj.weight: sum: 2546.014  dtype: BFloat16 shape: [5120,17920]
196: model.layers.38.mlp.gate_up_proj.weight: sum: -6134.1562  dtype: BFloat16 shape: [35840,5120]
197: model.layers.38.post_attention_layernorm.weight: sum: 47.6437  dtype: BFloat16 shape: [5120]
198: model.layers.38.self_attn.o_proj.weight: sum: 3502.9624  dtype: BFloat16 shape: [5120,5120]
199: model.layers.38.self_attn.qkv_proj.weight: sum: 4959.879  dtype: BFloat16 shape: [7680,5120]
200: model.layers.39.input_layernorm.weight: sum: 47.6586  dtype: BFloat16 shape: [5120]
201: model.layers.39.mlp.down_proj.weight: sum: 7300.0513  dtype: BFloat16 shape: [5120,17920]
202: model.layers.39.mlp.gate_up_proj.weight: sum: -9246.838  dtype: BFloat16 shape: [35840,5120]
203: model.layers.39.post_attention_layernorm.weight: sum: 47.7071  dtype: BFloat16 shape: [5120]
204: model.layers.39.self_attn.o_proj.weight: sum: 2975.9302  dtype: BFloat16 shape: [5120,5120]
205: model.layers.39.self_attn.qkv_proj.weight: sum: 5399.1167  dtype: BFloat16 shape: [7680,5120]
206: model.layers.4.input_layernorm.weight: sum: 47.565  dtype: BFloat16 shape: [5120]
207: model.layers.4.mlp.down_proj.weight: sum: -4974.919  dtype: BFloat16 shape: [5120,17920]
208: model.layers.4.mlp.gate_up_proj.weight: sum: 6113.4355  dtype: BFloat16 shape: [35840,5120]
209: model.layers.4.post_attention_layernorm.weight: sum: 47.626  dtype: BFloat16 shape: [5120]
210: model.layers.4.self_attn.o_proj.weight: sum: 3139.3357  dtype: BFloat16 shape: [5120,5120]
211: model.layers.4.self_attn.qkv_proj.weight: sum: 5102.535  dtype: BFloat16 shape: [7680,5120]
212: model.layers.5.input_layernorm.weight: sum: 47.5476  dtype: BFloat16 shape: [5120]
213: model.layers.5.mlp.down_proj.weight: sum: 6341.8745  dtype: BFloat16 shape: [5120,17920]
214: model.layers.5.mlp.gate_up_proj.weight: sum: 5587.1895  dtype: BFloat16 shape: [35840,5120]
215: model.layers.5.post_attention_layernorm.weight: sum: 47.6785  dtype: BFloat16 shape: [5120]
216: model.layers.5.self_attn.o_proj.weight: sum: 4280.1514  dtype: BFloat16 shape: [5120,5120]
217: model.layers.5.self_attn.qkv_proj.weight: sum: 2800.677  dtype: BFloat16 shape: [7680,5120]
218: model.layers.6.input_layernorm.weight: sum: 47.7139  dtype: BFloat16 shape: [5120]
219: model.layers.6.mlp.down_proj.weight: sum: 6957.9663  dtype: BFloat16 shape: [5120,17920]
220: model.layers.6.mlp.gate_up_proj.weight: sum: 6194.1855  dtype: BFloat16 shape: [35840,5120]
221: model.layers.6.post_attention_layernorm.weight: sum: 47.8007  dtype: BFloat16 shape: [5120]
222: model.layers.6.self_attn.o_proj.weight: sum: 2511.0842  dtype: BFloat16 shape: [5120,5120]
223: model.layers.6.self_attn.qkv_proj.weight: sum: 11261.535  dtype: BFloat16 shape: [7680,5120]
224: model.layers.7.input_layernorm.weight: sum: 47.6988  dtype: BFloat16 shape: [5120]
225: model.layers.7.mlp.down_proj.weight: sum: 2741.2383  dtype: BFloat16 shape: [5120,17920]
226: model.layers.7.mlp.gate_up_proj.weight: sum: 4584.136  dtype: BFloat16 shape: [35840,5120]
227: model.layers.7.post_attention_layernorm.weight: sum: 47.6654  dtype: BFloat16 shape: [5120]
228: model.layers.7.self_attn.o_proj.weight: sum: 3485.3408  dtype: BFloat16 shape: [5120,5120]
229: model.layers.7.self_attn.qkv_proj.weight: sum: 5009.4653  dtype: BFloat16 shape: [7680,5120]
230: model.layers.8.input_layernorm.weight: sum: 47.807  dtype: BFloat16 shape: [5120]
231: model.layers.8.mlp.down_proj.weight: sum: -12657.075  dtype: BFloat16 shape: [5120,17920]
232: model.layers.8.mlp.gate_up_proj.weight: sum: 2801.0295  dtype: BFloat16 shape: [35840,5120]
233: model.layers.8.post_attention_layernorm.weight: sum: 47.8077  dtype: BFloat16 shape: [5120]
234: model.layers.8.self_attn.o_proj.weight: sum: 1509.2021  dtype: BFloat16 shape: [5120,5120]
235: model.layers.8.self_attn.qkv_proj.weight: sum: 5331.6777  dtype: BFloat16 shape: [7680,5120]
236: model.layers.9.input_layernorm.weight: sum: 47.8249  dtype: BFloat16 shape: [5120]
237: model.layers.9.mlp.down_proj.weight: sum: 5878.0625  dtype: BFloat16 shape: [5120,17920]
238: model.layers.9.mlp.gate_up_proj.weight: sum: 6830.2754  dtype: BFloat16 shape: [35840,5120]
239: model.layers.9.post_attention_layernorm.weight: sum: 47.6564  dtype: BFloat16 shape: [5120]
240: model.layers.9.self_attn.o_proj.weight: sum: 4984.4287  dtype: BFloat16 shape: [5120,5120]
241: model.layers.9.self_attn.qkv_proj.weight: sum: -3516.8704  dtype: BFloat16 shape: [7680,5120]
242: model.norm.weight: sum: 47.6743  dtype: BFloat16 shape: [5120]
